core/helpers/hardhat/balance.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers } from 'ethers';
+import getContract, { getContractAddressByName, getErc20Contract } from '../../src/contracts';
+import BigNumber from '../../src/bignumber';
+import { EthereumProvider } from 'hardhat/types';
+import { TEST_NETWORK, HARDHAT_PUBLIC_KEY } from '../../helpers/constants';
+import { DAI_NUMBER_OF_DIGITS, MKR_NUMBER_OF_DIGITS, WAD_NUMBER_OF_DIGITS } from '../../src/constants/UNITS';
+import { CollateralConfig, CollateralType } from '../../src/types';
+import {
+    determineBalanceSlot,
+    overwriteUintMapping,
+    overwriteUintMappingInAddress,
+    overwriteUintTable,
+} from '../hardhat/slotOverwrite';
+import { runBalanceSlotDiscoveryLoopForERC20Token } from './slotOverwrite';
+
+export const addDaiToBalance = async (
+    daiAmount: BigNumber = new BigNumber(100000),
+    walletAddress: string = HARDHAT_PUBLIC_KEY
+) => {
+    const daiContract = await getContract(TEST_NETWORK, 'MCD_DAI', false);
+    await overwriteUintMapping('MCD_DAI', '0x2', walletAddress, daiAmount.shiftedBy(DAI_NUMBER_OF_DIGITS));
+    const daiBalanceHex = await daiContract.balanceOf(walletAddress);
+    const daiBalance = new BigNumber(daiBalanceHex._hex).shiftedBy(-DAI_NUMBER_OF_DIGITS);
+    console.info(`New DAI balance: ${daiBalance}`);
+};
+
+export const addMkrToBalance = async (
+    mkrAmount: BigNumber = new BigNumber(100000),
+    walletAddress: string = HARDHAT_PUBLIC_KEY
+) => {
+    const mkrContract = await getContract(TEST_NETWORK, 'MCD_GOV', false);
+    await overwriteUintMapping('MCD_GOV', '0x1', walletAddress, mkrAmount.shiftedBy(MKR_NUMBER_OF_DIGITS));
+    const mkrBalanceHex = await mkrContract.balanceOf(walletAddress);
+    const mkrBalance = new BigNumber(mkrBalanceHex._hex).shiftedBy(-MKR_NUMBER_OF_DIGITS);
+    console.info(`New MKR balance: ${mkrBalance}`);
+};
+
+export const setCollateralInVat = async (
+    collateralType: CollateralType,
+    collateralAmount: BigNumber,
+    provider?: EthereumProvider
+) => {
+    const value = collateralAmount.shiftedBy(WAD_NUMBER_OF_DIGITS);
+    const collateralTypeHex = ethers.utils.formatBytes32String(collateralType);
+    await overwriteUintTable('MCD_VAT', '0x4', collateralTypeHex, HARDHAT_PUBLIC_KEY, value, provider);
+};
+
+export const setCollateralInWallet = async (
+    collateralConfig: CollateralConfig,
+    collateralAmount: BigNumber,
+    provider?: EthereumProvider
```

Comment:
As suggested https://github.com/sidestream-tech/unified-auctions-ui/pull/499#discussion_r999132567
```suggestion
    address: string = HARDHAT_PUBLIC_KEY,
    provider?: EthereumProvider
```

In case we want to overwrite the join address balance for example